issue_15/script.js

```diff
@@ -0,0 +1,87 @@
+const showBtn = document.getElementById('js-show-btn');
+const modal = document.getElementById('js-modal');
+const closeBtn = document.getElementById('js-close-btn');
+const submitBtn = document.getElementById('js-submit-btn');
+
+
+function startLoading() {
+    const loadingImg = document.createElement('img');
+    loadingImg.src = 'loading-circle.gif';
+    loadingImg.alt = 'ローディング画像';
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoading() {
+    document.getElementById('js-loading').remove();
+}
+
+function renderList(resolvedData) {
+    const ul = document.getElementById('js-ul');
+    const fragment = document.createDocumentFragment();
+    for (const item of resolvedData) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.href = item.a;
+        a.textContent = item.text;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    ul.appendChild(fragment);
+}
+
+async function fetchData() {
+    try {
+        const URL = 'https://api.json-generator.com/templates/NluaaELSLhVe/data?access_token=zjrk7bmqgvptk4dcs8p8bhs5n9sk7m1ecs65gvq4';
+        const response = await fetch(URL);
+        if (!response.ok) {
+            console.error(`${response.status}:${response.statusText}`);
+        }
+        const result = await response.json();
+        return result.data;
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+async function init(num, text) {
+    startLoading();
+    let listData;
+    try {
+        console.log(num, text);
+        listData = await fetchData();
+        if (!listData) {
+            return;
+        }
+        renderList(listData);
+    } catch (error) {
+        console.error(error);
+    } finally {
+        removeLoading();
+    }
+}
+
+showBtn.addEventListener('click', () => {
+    modal.classList.add('show');
+    showBtn.classList.add('hide');
+});
+
+closeBtn.addEventListener('click', () => {
+    modal.classList.remove('show');
+    showBtn.classList.remove('hide');
+});
+
+submitBtn.addEventListener('click', () => {
+    const inputNumber = document.getElementById('js-input-number').value;
+    const inputText = document.getElementById('js-input-text').value;
+    if (inputNumber == '' || inputText == '') {
+        alert('未入力のフォームがあります');
+    } else {
+        init(inputNumber, inputText);
+        document.getElementById('js-form').reset();
+        modal.classList.remove('show');
+    }
+});
```

Comment:
IMO
This is correct but this way is simple and easier to understand for me. I will leave it up to you to modify it.

```suggestion
  if (inputNumber && inputText) {
    init(inputNumber, inputText);
    document.getElementById("js-form").reset();
    modal.classList.remove("show");
  } else {
    alert("未入力のフォームがあります");
  }
});
```

Reply:
I learned a lot about logical products, which I did not know.
Thank you:sparkles:
2ed279d8d18fd1ceb242e9c9e214099f15b1cf22